### ISSUE TITLE
Node 12 support dropped with #3245

### ISSUE THREAD

**Issue by mitchellbusby** [ipfs/js-ipfs#3270]
- **Version**:

IPFS 0.5+

- **Platform**:

N/A - not platform specific

- **Subsystem**:

N/A - ipfs core related

#### Severity:

High/Medium 

#### Description:

[merge-options](https://github.com/schnittstabil/merge-options) dropped support for Node 12 with their 3.x release, which IPFS just bumped up. This causes Node 12 install to fail.

Raising on the off chance this was missed -- it might be entirely intentional.

#### Steps to reproduce the error:
Try to install `ipfs` under node 12 -- the install of merge-options will fail.

**Comment by mitchellbusby** [comment]
Not sure how severe this actually is -- was a minor annoyance for me -- there may not actually be an intention to support node 12 at this stage, but I was surprised since it's the current LTS variant.

**Comment by achingbrain** [comment]
We run CI under node 12 which passes and I can't replicate this locally.  What's the error you are seeing?

**Comment by BlackGlory** [comment]
@achingbrain Node.js 12 LTS, first install `ipfs-http-client` v46.0.1, then upgrade it to v46.1.0, you will get:
```sh
error merge-options@3.0.1: The engine "node" is incompatible with this module. Expected version ">=14". Got "12.18.3"
Done in 10.69s.
Error: Found incompatible module.
    at MessageError.ExtendableBuiltin (/usr/share/yarn/lib/cli.js:721:66)
    at new MessageError (/usr/share/yarn/lib/cli.js:750:123)
    at checkOne (/usr/share/yarn/lib/cli.js:48058:11)
    at Object.check (/usr/share/yarn/lib/cli.js:48077:5)
    at /usr/share/yarn/lib/cli.js:7348:73
    at Generator.next (<anonymous>)
    at step (/usr/share/yarn/lib/cli.js:310:30)
    at /usr/share/yarn/lib/cli.js:321:13
```

**Comment by achingbrain** [comment]
Are you running with [engine-strict = true](https://docs.npmjs.com/misc/config#engine-strict)?

**Comment by achingbrain** [comment]
Ah, yarn.

**Comment by achingbrain** [comment]
It's a bit tedious, merge-options does have engines >= 14 in it's package.json since it's been rewritten to esm modules but is also published transpiled to common js and doesn't use any node 14 features, hence no CI failure.

**Comment by mitchellbusby** [comment]
Ahhh that makes sense - I was also using yarn 😅

**Comment by mitchellbusby** [comment]
@achingbrain is it possible to drop down to the previous version or is that a needed peer dep?

**Comment by mitchellbusby** [comment]
(To be clear this isn’t pressing for me - I’m stuck on older versions by way of OrbitDB)

**Comment by achingbrain** [comment]
Yes, I'll downgrade and publish a new version.

**Comment by achingbrain** [comment]
Fixed in `ipfs@0.50.1`
